crates/circuit/src/circuit_data.rs

```diff
@@ -995,6 +996,24 @@ impl CircuitData {
         self.param_table.clear();
     }
 
+    /// Counts the number of times each operation is used in the circuit.
+    ///
+    /// # Parameters
+    /// - `self` - A mutable reference to the CircuitData struct.
+    ///
+    /// # Returns
+    /// An IndexMap containing the operation names as keys and their respective counts as values.
+    pub fn count_ops(&mut self) -> IndexMap<String, usize, ::ahash::RandomState> {
+        let mut ops_count: IndexMap<String, usize, ::ahash::RandomState> = IndexMap::default();
+        for instruction in &self.data {
+            *ops_count
+                .entry(instruction.op.view().name().to_string())
+                .or_insert(0) += 1;
+        }
```

Comment:
I'm not actually suggesting you make this change, as I think a for loop is probably more legible. But I was curious if you could do this with just an iterator which you can do like this:

```suggestion
        let mut ops_count =
            self.data
                .iter()
                .fold(IndexMap::default(), |mut ops_count, instruction| {
                    *ops_count.entry(instruction.op.name()).or_insert(0) += 1;
                    ops_count
                });
```

I don't think there is any advantage to writing it like this.  So it's more just an interesting musing on how to use functional programming style iterators for this.

Reply:
I'll test it out

Reply:
seemed to work